atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/invariant/SerializableInvariant.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.workload.invariant;
+
+import com.palantir.atlasdb.workload.store.TableAndWorkloadCell;
+import com.palantir.atlasdb.workload.transaction.InMemoryTransactionReplayer;
+import com.palantir.atlasdb.workload.transaction.witnessed.InvalidWitnessedTransaction;
+import com.palantir.atlasdb.workload.transaction.witnessed.InvalidWitnessedTransactionAction;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedDeleteTransactionAction;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedReadTransactionAction;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedTransactionActionVisitor;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedWriteTransactionAction;
+import com.palantir.atlasdb.workload.workflow.WorkflowHistory;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import one.util.streamex.StreamEx;
+
+public enum SerializableInvariant implements Invariant<List<InvalidWitnessedTransaction>> {
+    INSTANCE;
+
+    @Override
+    public void accept(
+            WorkflowHistory workflowHistory, Consumer<List<InvalidWitnessedTransaction>> invalidWitnessedTransactions) {
+        SerializableInvariantVisitor visitor = new SerializableInvariantVisitor();
+        List<InvalidWitnessedTransaction> transactions = StreamEx.of(workflowHistory.history())
```

Comment:
I know previously we discussed that we opt to use `stream()` rather than StreamEx, but I decided to use it here for the `mapPartial` feature, hope that's ok!